LibSWBF2/Wrappers/Terrain.cpp

```diff
@@ -236,8 +300,55 @@ namespace LibSWBF2::Wrappers
 		return p_Terrain->p_Name->m_Text;
 	}
 
+
+	void Terrain::GetHeightMap(uint32_t& dim, uint32_t& dimScale, float_t*& heightData) const
+    {
+        auto info = p_Terrain -> p_Info;
+
+        float_t gridSize     = (float_t) info -> m_GridSize;
+		float_t gridUnitSize = (float_t) info -> m_GridUnitSize;
+
+		float_t maxY = info -> m_HeightCeiling;
+       	float_t minY = info -> m_HeightFloor;
+
+       	float_t halfLength = gridSize * gridUnitSize / 2.0f;
+       	float_t maxZ = halfLength, minZ = -halfLength;
+       	float_t maxX = halfLength, minX = -halfLength;
+
+        Vector3 *vertexBuffer = m_Positions.GetArrayPtr();
+        uint32_t numVerts = m_Positions.Size();
+
+        int heightsLength = (int) gridSize * gridSize;
+		heightData = new float_t[heightsLength]();
+		dim = (uint32_t) info -> m_GridSize;
+		dimScale = (uint32_t) info -> m_GridUnitSize;
+
+        for (int i = 0; i < numVerts; i++)
+        {
+        	Vector3& curVert = vertexBuffer[i];
+        	float_t xFrac = (curVert.m_X - minX)/(maxX - minX);
+        	float_t yFrac = (curVert.m_Y - minY)/(maxY - minY);
+        	float_t zFrac = (curVert.m_Z - minZ)/(maxZ - minZ);
+
+        	int uIndex = (int) (xFrac * gridSize + .00001f);
+        	int vIndex = (int) (zFrac * gridSize + .00001f);
+            int dataIndex = uIndex + vIndex * (int) gridSize;
+
+            if (uIndex < (int) gridSize && vIndex < (int) gridSize)
+            {
```

Comment:
This could be cleaned up a bit, will get to that soon

Reply:
Just keep in mind that there are patches with an altered vertex buffer and custom index buffer (baked in terrain holes). So you cannot assume that, given a patch edge of 8, each patch always has 64 vertices. This is actually a bug in sleepkillers unmunge

Reply:
Not sure if that bug effects this code, since it just transforms physical location to logical index in the full heightmap, which has enough space for the whole grid (determined by gridUnitSize/gridSize).  But that is a big issue for how I deal with the holes texture Unity requires...